src/connector/src/sink/kafka.rs

```diff
@@ -403,7 +403,16 @@ struct KafkaPayloadWriter<'a> {
     config: &'a KafkaConfig,
 }
 
-pub type KafkaSinkDeliveryFuture = impl TryFuture<Ok = (), Error = SinkError> + Unpin + 'static;
+mod opaque_type {
```

Comment:
Is this still necessary as we won't bump to `> 2024-06-12`?

Reply:
Not necessary, but I think it doesn't hurt.